.github/workflows/unit_tests.yml

```diff
@@ -57,21 +57,17 @@ jobs:
           # - without socks support
           # - without http2 support
           TO_TEST="test_no_passport.py or test_datetime.py or test_defaults.py or test_jobqueue.py or test_applicationbuilder.py or test_ratelimiter.py or test_updater.py or test_callbackdatacache.py or test_request.py"
-          pytest -v --cov -k "${TO_TEST}"
-          # Rerun only failed tests (--lf), and don't run any tests if none failed (--lfnf=none)
-          pytest -v --cov --cov-append -k "${TO_TEST}" --lf --lfnf=none --junit-xml=.test_report_no_optionals.xml
```

Comment:
why don't we run the test suite again to reduce CI failures? I didn't get the explanation in the commit message of https://github.com/python-telegram-bot/python-telegram-bot/pull/4487/commits/1f2144a790661acb4dff3c1a545e65226728ae58

Reply:
TBH, I don't have a solid fact-based argumentation 😅 IIRC #3631/#3626 where already more of a "let's see if this helps". Properly analyzing & addressing tests that tend to fail is more robust in the long run, I'd guess. Since we can now make that analysis, we don't necessarily need to water down the analysis.

Reply:
I think that some of the tests in the CI were simply failing due to race conditions or some other hard to debug reason, which is why I proposed that. Since that time we might have fixed many such tests. I suppose now we could remove this and check again.